### PR TITLE
dockercompose: allow docker-compose build to run in parallel, even though docker-compose up cannot

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @jazzdan,

Please review the following commits I made in branch nicks/ch8819:

b7e50dc5ebc8cb7d5b5475c2c5c999d3e3d7b1c0 (2020-08-07 11:29:40 -0400)
dockercompose: allow docker-compose build to run in parallel, even though docker-compose up cannot
Fixes https://github.com/tilt-dev/tilt/issues/3665

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
wahhhh

**Comment by maiamcc** [comment]
> wahhhh

...Oh dear. You okay there, @nicks?

**Comment by nicks** [comment]
Just my response to taking DM off the reviewer list

File: internal/dockercompose/client.go [hunk 1]
```diff
@@ -44,34 +44,46 @@ func NewDockerComposeClient(env docker.LocalEnv) DockerComposeClient {
...
+		cmd.Stderr = stderr
+		err := cmd.Run()
+		if err != nil {
+			return FormatError(cmd, nil, err)
+		}
+	}
+
+	// docker-compose up is not thread-safe, because network operations are non-atomic. See:
+	// https://github.com/tilt-dev/tilt/issues/2817
+	// TODO(nick): It might make sense to use a CondVar so that we can log a message
```
Comment (maiamcc) [inline internal/dockercompose/client.go:70]: (maybe worth adding a note that `docker-compose build` (as called above) CAN run in parallel, so we don't need to lock the mutex for it?)

**Comment by maiamcc** [comment]
RIP dan miller :(

File: internal/dockercompose/client.go [hunk 2]
```diff
@@ -44,34 +44,46 @@ func NewDockerComposeClient(env docker.LocalEnv) DockerComposeClient {
...
+	// when we're waiting on another build...
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	runArgs := append([]string{}, genArgs...)
+	runArgs = append(runArgs, "up", "--no-deps", "-d")
```
Comment (maiamcc) [inline internal/dockercompose/client.go:75]: (I assume `docker-compose up` is smart enough to not build the image if it was just built with `docker-compose build`?)

Comment (nicks) [reply internal/dockercompose/client.go:75]: ya, i believe it's pretty smart about that, but i can set --no-build to emphasize this point

File: internal/dockercompose/client.go (hunk 1 above)
Comment (nicks) [reply internal/dockercompose/client.go:73]: done

**Pull request merged**